include/alpaka/vec/Vec.hpp

```diff
@@ -555,6 +555,66 @@ namespace alpaka
             TSize m_data[TDim::value == 0u ? 1u : TDim::value];
         };
 
+        //-----------------------------------------------------------------------------
+        //! This is a conveniance method to have a out-of-class factory method even though the out-of-class version is not supported by all compilers.
+        //! This has the draw-back, that it requires the TSize parameter even though it should not be necessary.
+        //-----------------------------------------------------------------------------
+        ALPAKA_NO_HOST_ACC_WARNING
+        template<
+            typename TDim,
+            typename TSize,
+            template<std::size_t> class TTFnObj,
+            typename... TArgs>
+        ALPAKA_FN_HOST_ACC auto createVecFromIndexedFn2(
```

Comment:
Could we find a better name for `createVecFromIndexedFn2` e.g. `indexedFnToVec`or only `vecFromIndexed`. 
The point that a vector is created should be clear from the result type.